### ISSUE TITLE
Ability to start with Steam.

### ISSUE THREAD

**Issue by thakyZ** [rumangerst/SilVerPLuM#4]
I would like to have a way to start the game with Steam, either with `steam://######` or with a shortcut, but the first option may be better for other systems.

I know how to make Stardew Valley open with SMAPI on Steam without doing a non-library game. (which is if you would like to know so that you can detect if the person is using SMAPI, or not, and warn them accordingly: `"C:\Program Files (x86)\Steam\steamapps\common\Stardew Valley\StardewModdingAPI.exe" %command%` That is for Windows, but I haven't tested it on Linux. (You would set that line of text for the launch options)).

This will allow the player to get Steam achievements and have Steam overlay while in the game.

**NOTE:** I don't know how to do the launch options for STORM but I would guess it is the same thing.

**Comment by rumangerst** [comment]
I just tried it the SMAPI approach out: http://imgur.com/a/26t9u

At least the overlay works OOTB (on Linux). This "%command%" thing is a mystery for me. I added some additional log-entry that fetches those arguments and for me this arguments just was STEAM_RUNTIME=0 (set by my Linux distribution). OK, the icons in SilVerPLuM are broken (invisible) for some reason (probably due to Steam fiddling around with the library configuration) 

About the achievements: I don't know how Steam gets them. Do they work with SMAPI? Could be handled by the game itself if Steam knows it runs (?)

**Comment by thakyZ** [comment]
What I have for my library when I launch it with the `%command%` part is that achievements do work, steam just has to think it is running with Stardew Valley, whether it is SMAPI or Stardew Valley it's self.

**Comment by rumangerst** [comment]
Do achievements work for you if you just do `"<path to Silverplum>" %command%` ? I investigated %command% on a Windows VM. It's just the SDV exe. 
There's a possibility that it just works if you start SDV/SMAPI/whatever within Silverplum.

**Comment by thakyZ** [comment]
But don't you have the push the commands to the game after you start Silverplum, and then start the game?

**Comment by rumangerst** [comment]
What commands should be pushed to the game? (If I understood you correctly)

**Comment by thakyZ** [comment]
Well if Steam starts the game with `STEAM_RUNTIME=0` when `%command%` gets called while you launch SilVerPLuM then when you press play on SilVerPLuM. Would SilVerPLuM execute `StardewValley.exe STEAM_RUNTIME=0` automaticly, or would you have to program that into SilVerPLuM just to do that?

**Comment by rumangerst** [comment]
Ahh. I tested it again under Linux and I've somehow overlooked something:

`[INFO]	program	program	arguments	/home/ruman/Project/SilVerPLuM/build-SilVerPLuM-Desktop-Debug/SilVerPLuM "/home/ruman/.local/share/Steam/steamapps/common/Stardew Valley/StardewValley" STEAM_RUNTIME=0`

And the [Arch Wiki](https://wiki.archlinux.org/index.php/steam#Launching_games_with_custom_commands) says that %command% is just replaced by the program that is usually started.

> Would SilVerPLuM execute StardewValley.exe STEAM_RUNTIME=0 automaticly, or would you have to program that into SilVerPLuM just to do that?
I could do this if I would include this behavior, but then there would be no DLL-mods. And Silverplum doesn't know which profile should be chosen and more problems.

It should work to just start SDV as usual within Silverplum. I cannot say it for sure, but its fact that SDV connects to Steam via a library. It is then plausible that if Steam thinks a game is open (Silverplum *will* stay open until you close it manually), SDV's steam library can connect to Steam and handle everything.

I also looked through the code of SMAPI and nowhere commandline arguments are picked up there (Program.cs and Environment.GetCommandLineArgs). Seems to just ignore the command from Steam.
